### PR TITLE
GH-35130: [Docs] Document how to become a collaborator to get triage role

### LINKED ISSUE #35130: [Docs] Document how to become collaborator on the Arrow project
### Describe the enhancement requested

With the migration to GitHub we have started using the collaborator role which allows us to give triage role to some users.
I was myself a collaborator before being a committer. This was done in order to be able to label issues, assign them, etcetera.

The role is documented here from the INFRA wiki: https://cwiki.apache.org/confluence/display/INFRA/Git+-+.asf.yaml+features#Git.asf.yamlfeatures-AssigningexternalcollaboratorswiththetriageroleonGitHub

We should document how are we giving triage role so there's a clear way of giving the role to new users.

### Component(s)

Documentation

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Since we migrated to GitHub we have been using the collaborator role to allow some contributors to start helping with a triage role.
There has been some discussions on making more clear on the process to become a collaborator.

### What changes are included in this PR?

Add some documentation on how to propose / become a collaborator.

### Are these changes tested?

Will trigger the preview-docs.

### Are there any user-facing changes?

No
* Closes: #35130

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit preview-docs

**Comment by raulcd** [comment]
The changes can be seen here: http://crossbow.voltrondata.com/pr_docs/36445/developers/reviewing.html#collaborators

File: docs/source/developers/reviewing.rst [hunk 1]
```diff
@@ -295,3 +295,21 @@ In addition, we use the following labels to indicate priority:
...
+
+Collaborators
+=============
+
+The collaborator role allows users to be given triage role to be able to help triaging
+issues. The role allows users to label and assign issues.
+
+A user can ask to become a collaborator or can be proposed by another community member
+when they have been collaborating in the project for a period of time. Collaborations
+can be but are not limited to: creating Pull requests, answering questions, creating
```
Comment (jorisvandenbossche) [inline docs/source/developers/reviewing.rst:307]:
```suggestion
can be but are not limited to: creating PRs, answering questions, creating
```

(we are already using this abbreviation earlier in the document)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks for writing this up!

**Review by kou — APPROVED** [review]
+1

File: docs/source/developers/reviewing.rst [hunk 2]
```diff
@@ -295,3 +295,21 @@ In addition, we use the following labels to indicate priority:
...
+The collaborator role allows users to be given triage role to be able to help triaging
+issues. The role allows users to label and assign issues.
+
+A user can ask to become a collaborator or can be proposed by another community member
+when they have been collaborating in the project for a period of time. Collaborations
+can be but are not limited to: creating PRs, answering questions, creating
+issues, reviewing PRs, etcetera.
+
+In order to propose someone as a collaborator you must create a PR adding the user to
+the collaborators list on `.asf.yaml <https://github.com/apache/arrow/blob/main/.asf.yaml>`_..
```
Comment (kou) [inline docs/source/developers/reviewing.rst:311]:
```suggestion
the collaborators list on `.asf.yaml <https://github.com/apache/arrow/blob/main/.asf.yaml>`_.
```

**Pull request merged**